File: BAEKJOON_SET/Set_Jonghwan.py
s = 0
tc_num = int(input().strip())
for i in range(tc_num):
    inputs = input().strip().split()
    if(len(inputs) == 1):
        op = inputs
    else:
        op = inputs[0]
        data = int(inputs[1])
    if(op == 'add'):
        s = s | (1<<data)
    elif(op == 'check'):
        res = s & (1<<data)
        print(res>>data)
    elif(op == 'toggle'):
        s = s ^ (1<<data)
    elif(op == 'remove'):
        s = s & ~(1<<i)
    elif(op == 'all'):
        s = 1048575
    elif(op == 'empty'):
        s = 0

File: BAEKJOON_SET/Set_Jonghwan.c
#include <stdio.h>

int main()
{
	int tc_num = 0;
	char op[7];
	int data = 0;
	int S = 0;
	int check = 0;
	scanf("%d",&tc_num);
	for(int i=0;i<tc_num;i++)
	{
		scanf(" %s %d",&op,&data);
		//printf("%s %d",op,data);
		switch(op[0]){
			case 'a':
				if(op[1] == 'd')
				{
					S = S | (1<<data);
				}
				else
				{
					S = (1 << 21)-1;
				}
				break;
			case 'c':
				check = (S & (1<<data));
				if(check > 0)
					printf("1\n");
				else
					printf("0\n");
				break;
			case 'r':
				S = S & ~(1<<data);
				break;
			case 't':
				S = S ^ (1<<data);
				break;
			case 'e':
				S = 0;
				break;
			default:
				break;
		} 
	}
}
